rasa/nlu/training_data/message.py

```diff
@@ -24,16 +26,16 @@
 class Message:
     def __init__(
         self,
-        text: Text,
+        text: Optional[Text] = "",
         data: Optional[Dict[Text, Any]] = None,
         output_properties: Optional[Set] = None,
         time: Optional[Text] = None,
         features: Optional[List["Features"]] = None,
         **kwargs,
     ) -> None:
-        self.text = text
         self.time = time
-        self.data = data if data else {}
+        self.data = data.copy() if data else {}
+        self.data.update({TEXT: text})
```

Comment:
```suggestion
        self.data[TEXT] = text
```

Reply:
I'll leave it as is for now, because when we merge the `removal of text` PR these lines will be gone anyway.